src/QueryM/DatumCacheWsp.purs

```diff
@@ -104,24 +106,56 @@ instance DecodeAeson GetDatumsByHashesR where
   decodeAeson r =
     let
       decodeDatumArray
-        :: Aeson -> Either JsonDecodeError (Map DataHash Datum)
+        :: Aeson -> Either JsonDecodeError (Map DataHash (Either String Datum))
       decodeDatumArray =
         caseAesonArray (Left $ TypeMismatch "expected array")
           $ (map Map.fromFoldable) <<< traverse decodeDatum
 
+      decodeEmptyDatums obj =
+        pure Map.empty *>
+          ( caseAesonArray (Left $ TypeMismatch "expected array")
+              $ identity <<< traverse decodeEmptyDatum
+          ) obj
+
+      decodeEmptyDatum
+        :: Aeson -> Either JsonDecodeError Unit
+      decodeEmptyDatum aes =
+        if stringifyAeson aes == "{}" then
+          pure unit
+        else
+          Left $ TypeMismatch "expected empty JSON"
+
       decodeDatum
-        :: Aeson -> Either JsonDecodeError (DataHash /\ Datum)
-      decodeDatum = caseAesonObject (Left $ TypeMismatch "expected object")
-        $ \o -> (/\) <$> map wrap (o .: "hash") <*>
-            (decodeAeson =<< o .: "value")
+        :: Aeson -> Either JsonDecodeError (DataHash /\ Either String Datum)
+      decodeDatum obj = caseAesonObject (Left $ TypeMismatch "expected object")
+        (\o -> (/\) <$> map wrap (o .: "hash") <*> (decodeValueOption obj))
+        obj
+
+      decodeValueOption
+        :: Aeson -> Either JsonDecodeError (Either String Datum)
+      decodeValueOption aes =
+        do
+          options <- (pure <$> getNestedAeson aes [ "value", "Right" ])
+            <|> (Left <$> getNestedAeson aes [ "value", "Left" ])
+          case options of
+            Left x -> pure $ Left $ show x
+            Right x -> pure <$> decodeAeson x
+
       datumsFound =
         map GetDatumsByHashesR <<< decodeDatumArray =<< getNestedAeson
           r
           [ "DatumsFound", "value" ]
+
+      datumsEmpty =
+        ( decodeEmptyDatums =<< getNestedAeson
+            r
+            [ "DatumsFound", "value" ]
+        ) $> GetDatumsByHashesR Map.empty
+
       datumsNotFound =
         getNestedAeson r [ "DatumsNotFound" ] $> GetDatumsByHashesR Map.empty
     in
-      datumsFound <|> datumsNotFound
+      datumsFound <|> datumsEmpty <|> datumsNotFound
```

Comment:
There are two cases, the first is `The request of datums was made with an empty list of Hashes`, and the second is `No one of the datums was found`.

Reply:
If you've made a request with an empty array of hashes, won't `datumsFound` succeed? Because it traverses the response array, which because it's empty will run no effects and thus succeed. Maybe it would help me to understand if I saw example requests + responses

Reply:
When I was writing the ODC update I was trying to avoid API changes, but now it seems that decision wasn't good in the end. As a result, we have some complexity here, I would make an issue for a change so that we would see the following responses.

With the new proposal we would get in case of an empty list : 

`{"DatumsFound" : {"value": [  ] } } `

And if we can't find some datums : 

`{"DatumsFound" : {"value": [ {"hash":"SomeHash", "value":{"Left":"DatumNotFound"}} ... ] } } `
 


The current code in case of an empty list of datums returns :  

`{"DatumsFound" : {"value": [ {} ] } } `

And If we can't find any of the datums also returns : 

`{"DatumsFound" : {"value": [ {} ] } } `

Reply:
the issues is here https://github.com/mlabs-haskell/ogmios-datum-cache/issues/125

Reply:
The correction is waiting for merge but this pr is now using it.